data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostContributionAndProofs.java

```diff
@@ -0,0 +1,102 @@
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_BAD_REQUEST;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_INTERNAL_ERROR;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_OK;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.TAG_EXPERIMENTAL;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.google.common.base.Throwables;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.ValidatorDataProvider;
+import tech.pegasys.teku.api.schema.altair.SignedContributionAndProof;
+import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class PostContributionAndProofs implements Handler {
+
+  public static final String ROUTE = "/eth/v1/validator/contribution_and_proofs";
+
+  private final ValidatorDataProvider provider;
+  private final JsonProvider jsonProvider;
+
+  public PostContributionAndProofs(final DataProvider provider, final JsonProvider jsonProvider) {
+    this(provider.getValidatorDataProvider(), jsonProvider);
+  }
+
+  public PostContributionAndProofs(
+      final ValidatorDataProvider provider, final JsonProvider jsonProvider) {
+    this.provider = provider;
+    this.jsonProvider = jsonProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.POST,
+      summary = "Publish contribution and proofs",
+      tags = {TAG_EXPERIMENTAL},
+      requestBody =
+          @OpenApiRequestBody(
+              content = {@OpenApiContent(from = SignedContributionAndProof.class, isArray = true)}),
+      description =
+          "Verifies given sync committee contribution and proofs and publishes on appropriate gossipsub topics.",
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            description = "Successfully published contribution and proofs."),
+        @OpenApiResponse(status = RES_BAD_REQUEST, description = "Invalid parameter supplied."),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR, description = "Beacon node internal error.")
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    try {
+      final SignedContributionAndProof[] signedContributionAndProofs =
+          jsonProvider.jsonToObject(ctx.body(), SignedContributionAndProof[].class);
+
+      provider
+          .sendContributionAndProofs(asList(signedContributionAndProofs))
+          .finish(error -> handleError(ctx, error));
```

Comment:
This doesn't feel right - either the response will be sent as successful before this future finishes or we'll never tell it if the future completes successfully.  I feel like there's a `ctx.async` type call missing here that should be used instead of `.finish`